src/data/adapter.h

```diff
@@ -234,22 +234,22 @@ class ArrayAdapterBatch : public detail::NoMetaInfo {
   class Line {
     ArrayInterface array_interface_;
     size_t ridx_;
+
    public:
     Line(ArrayInterface array_interface, size_t ridx)
         : array_interface_{std::move(array_interface)}, ridx_{ridx} {}
 
     size_t Size() const { return array_interface_.num_cols; }
 
     COOTuple GetElement(size_t idx) const {
-      return {ridx_, idx, array_interface_.GetElement(idx)};
+      return {ridx_, idx, array_interface_.GetElement(ridx_, idx)};
```

Comment:
In the adapter pattern created by @RAMitchell , a "line" is just a row, `idx` means the column in that row.  So we are accessing the same element.

Reply:
That means, `idx` is local to the line.

Reply:
In that case, shouldn't it be `array_interface_.GetElement(ridx_, 0)`?

Reply:
Em .. why?

Reply:
If both `ridx_` and `idx` are nonzero, then `array_interface_.GetElement(ridx_, idx)` would access something like `ridx_ * strides[0] + idx` which is larger than `idx`, right?

Reply:
... Unless the `array_interface_` is now pointing to something different...

Reply:
> rray_interface_.GetElement(ridx_, idx) would access something like ridx_ * strides[0] + idx which is larger than idx, right?

Right.  Sorry for the confusion.  Let me reprase it into:
```
output = values[row_index * row_stride + column_idx]
```

Reply:
This is the general pattern for indexing array: https://numpy.org/doc/stable/reference/arrays.ndarray.html#internal-memory-layout-of-an-ndarray

Reply:
The old code says `array_interface_.GetElement(idx)`, and the new code says `array_interface_.GetElement(ridx_, idx)`. I still don't understand how these two are semantically equivalent, since we have no reason to believe `row_stride == 0`.